### PR TITLE
Moved frontend/base/resourcebundles to its own module

### PR DESCRIPTION
**Pull request by MaciejG604**
…its own module

### CODE REVIEW

**Comment by PawelLipski** [comment]
Nit: as a general rule, pls use infinitive (_move_) rather than past participle (_moved_) in commit messages/PR titles etc.: https://cbea.ms/git-commit/#imperative (aka _If applied, this commit will..._)

File: frontend/base/src/main/java/com/virtuslab/gitmachete/frontend/icons/MacheteIcons.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import com.intellij.openapi.util.IconLoader;
+
+/**
+ * Icons must be in package named "icons" based on
+ * https://www.jetbrains.org/intellij/sdk/docs/reference_guide/work_with_icons_and_images.html
+ */
+public final class MacheteIcons {
+  private MacheteIcons() {}
+
+  public static final Icon DISCOVER = loadIcon("/icons/macheteLogoIcon.svg");
```
Comment (PawelLipski) [inline frontend/base/src/main/java/com/virtuslab/gitmachete/frontend/icons/MacheteIcons.java:14]: While you're here: all paths have `/icons/` prefix and `.svg` extension... maybe just replace `path` param in `loadIcon` with `basename` and add prefix/suffix automatically

File: frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileType.java [hunk 2]
```diff
@@ -3,9 +3,9 @@
 import javax.swing.Icon;
 
 import com.intellij.openapi.fileTypes.LanguageFileType;
-import icons.MacheteIcons;
```
Comment (PawelLipski) [inline frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileType.java:6]: @mkondratek wasn't there a requirement for these icons to reside in `icons` package? if so, it should be documented in package-info.java

@MaciejG604 are the icons still rendered correctly everywhere in the plugin?

File: scripts/prohibit-kotlin-undefined-property-keys [hunk 3]
```diff
@@ -1,11 +1,11 @@
- #!/usr/bin/env bash
+#!/usr/bin/env bash
```
Comment (PawelLipski) [inline scripts/prohibit-kotlin-undefined-property-keys:1]: Whoops good spot 😅

File: scripts/validate-icons [hunk 4]
```diff
@@ -5,11 +5,11 @@ set -e -o pipefail -u
 self_dir=$(cd "$(dirname "$0")" &>/dev/null; pwd -P)
 source "$self_dir"/utils.sh
 
-ICONS_PATH='frontend/icons/src/main/resources'
```
Comment (PawelLipski) [inline scripts/validate-icons:8]: While you're here... this is an unexported constant, I'd rather expect it to be in lower_snake_case. UPPER_SNAKE_CASE is conventionally restricted to exported vars

File: frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileType.java (hunk 2 above)
Comment (PawelLipski) [reply frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteFileType.java:6]: Nvm, see `frontend/icons/src/main/java/icons/MacheteIcons.java` below

File: frontend/icons/src/main/java/icons/MacheteIcons.java [hunk 5]
```diff
@@ -1,50 +0,0 @@
-package icons;
-
-import javax.swing.Icon;
-
-import com.intellij.openapi.util.IconLoader;
-
-/**
- * Icons must be in package named "icons" based on
- * https://www.jetbrains.org/intellij/sdk/docs/reference_guide/work_with_icons_and_images.html
```
[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontend/icons/src/main/java/icons/MacheteIcons.java:9]: _Starting with 2021.2, *Icons class is not required to be located in icons package but can use plugin's package: icons.MyIcons → my.plugin.MyIcons._

We still support 2 versions earlier than 2021.2.

Actually, the change with moving the class out of `icons` package can be done on https://github.com/VirtusLab/git-machete-intellij-plugin/pull/954

File: settings.gradle [hunk 6]
```diff
@@ -18,11 +18,11 @@ include 'branchLayout:api'
...
 include 'backend:impl'
-include 'frontend:icons'
 include 'frontend:base'
 include 'frontend:actions'
 include 'frontend:file'
 include 'frontend:graph:api'
 include 'frontend:graph:impl'
 include 'frontend:ui:api'
 include 'frontend:ui:impl'
+include 'frontend:resourcebundles'
```
Comment (PawelLipski) [inline settings.gradle:28]: Nit: as per the comment above, pls keep the projects in [topological order](https://pl.wikipedia.org/wiki/Sortowanie_topologiczne)

File: frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMacheteVisibilityPredicate.java [hunk 7]
```diff
@@ -1,17 +1,24 @@
...
+import java.util.function.Predicate;
+
 import com.intellij.openapi.project.Project;
 import com.intellij.util.NotNullFunction;
 import com.intellij.vcs.log.impl.VcsProjectLog;
 import lombok.CustomLog;
 
 @CustomLog
-public class GitMacheteVisibilityPredicate implements NotNullFunction<Project, Boolean> {
+public class GitMacheteVisibilityPredicate implements NotNullFunction<Project, Boolean>, Predicate<Project> {
```
[the remark this answers is not in the document]
Comment (MaciejG604) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMacheteVisibilityPredicate.java:11]: This is also highlighted as an error on develop/.

[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMacheteVisibilityPredicate.java:11]: Highlighted indeed... but it doesn't break compatibility as such. @mkondratek if you could elaborate

[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMacheteVisibilityPredicate.java:11]: See https://github.com/VirtusLab/git-machete-intellij-plugin/pull/954/files#diff-e29ca242413d85a05c2634490cfcf4a5821ee1122e0cf7dfeedfd8cd71d7aac8

[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/root/GitMacheteVisibilityPredicate.java:11]: So this change can be included... but will be reverted anyway once PR #954 is merged

File: frontend/resourcebundles/build.gradle [hunk 8]
```diff
@@ -0,0 +1,3 @@
+jetbrainsAnnotations()
+
+addIntellijToCompileClasspath(withGit4Idea: false)
```
Comment (PawelLipski) [inline frontend/resourcebundles/build.gradle:3]: Actually... going even further, is IntelliJ even needed on the classpath of `:frontend:resourcebundles` at all? 🤔

File: frontend/icons/src/main/java/icons/MacheteIcons.java [hunk 9]
```diff
@@ -11,40 +11,40 @@
 public final class MacheteIcons {
```
[the remark this answers is not in the document]
Comment (MaciejG604) [reply frontend/icons/src/main/java/icons/MacheteIcons.java:11]: Stays a separete module until we drop the support for IntelliJ 2020.3 nad 2021.1 so PR #954

[the remark this answers is not in the document]
Comment (PawelLipski) [reply frontend/icons/src/main/java/icons/MacheteIcons.java:11]: Makes sense, not to complicate things... lemme comment on that on #954 then

**Pull request merged**